app/services/nomis/custody/api.rb

```diff
@@ -17,26 +20,44 @@ def initialize
       def fetch_nomis_staff_details(username)
         route = "/custodyapi/api/nomis-staff-users/#{username}"
         response = @custodyapi_client.get(route)
-        api_deserialiser.deserialise(Nomis::StaffDetails, response)
+
+        ApiObjectResponse.new(api_deserialiser.deserialise(Nomis::StaffDetails, response))
       end
 
-      def get_offenders(prison)
-        route = "/custodyapi/api/offenders/prison/#{prison}?page=0&size=10"
-        response = @custodyapi_client.get(route)
-        response['_embedded']['offenders'].map do |offender|
+      # rubocop:disable Metrics/MethodLength
+      def get_offenders(prison, page = 0)
+        route = "/custodyapi/api/offenders/prison/#{prison}?page=#{page}&size=10"
+        page_meta = nil
+
+        response = @custodyapi_client.get(route) { |data|
```

Comment:
Just noticed you are passing a block :-)

Reply:
I switched it around, so rather than telling the get() method when to fail, we instead just inject the block in between when the call has completed and before it has returned so we can abort if necessary.  Less direct impact on the client.

Reply:
Good idea